### PR TITLE
Add a separate devops page

### PR DESCRIPTION
**Pull request by emmastephenson**
Fixes [#481](https://github.com/CDCgov/phdi/issues/481)

Part of the handoff plan work. This moves the CI/CD documentation to a separate page, and adds a suggested release process.

NOTE: I just guessed at the releases/release cadence suggestion, please feel free to make suggestions for improvement!

### CODE REVIEW

File: docs/devops.md [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+### Continuous Deployment (CD)
+
+A separate CD pipeline is configured for each Azure environment we deploy to. Each of these pipelines is defined in a YAML file starting with "deploy" in the `workflows/` directory (e.g. `phdi-azure/.github/deployment.yaml`). Generally, these pipelines run every time code is merged into the `main` branch of the repository. However, additional dependencies can be specified. For example, a successful deployment to a development environment could be required before deploying to a production environment. When these pipelines run Terraform looks for differences between the infrastructure that is specified on a given branch of this repository and what is currently deployed to a given environment in the Azure resource group. If differences are detected, they are resolved by making changes to Azure resources to bring them alignment with the repository. In order to grant the GitHub repository permission to make these changes, follow [these instructions](https://learn.microsoft.com/en-us/azure/active-directory/develop/workload-identity-federation-create-trust?pivots=identity-wif-apps-methods-azp#github-actions) to authenticate it with Azure.
```
Comment (nickclyde) [inline docs/devops.md:20]:
```suggestion
Our deployment pipeline is defined in the YAML file `phdi-azure/.github/workflows/deployment.yaml`. Generally, this pipeline runs every time code is merged into the `main` branch of the repository, deploying to the `dev` environment. The pipeline can also be triggered manually to deploy to other environments, and a successful deployment to a development environment could be required before deploying to a production environment. When the pipeline runs, Terraform looks for differences between the infrastructure that is specified on a given branch of this repository and what is currently deployed to a given environment in the Azure resource group. If differences are detected, they are resolved by making changes to Azure resources to bring them alignment with the repository. In order to grant the GitHub repository permission to make these changes, follow [these instructions](https://learn.microsoft.com/en-us/azure/active-directory/develop/workload-identity-federation-create-trust?pivots=identity-wif-apps-methods-azp#github-actions) to authenticate it with Azure.
```

**Review by nickclyde — CHANGES_REQUESTED** [review]
Looks good! Just one change:

File: docs/devops.md (hunk 1 above)
Comment (emmastephenson) [reply docs/devops.md]: Thanks for this! Much better 👍

**Review by nickclyde — APPROVED** [review]
:shipit:

**Pull request merged**